### PR TITLE
net.nim: add support for Unix sockets

### PR DESCRIPTION
**Pull request by zielmicha**
Adds support for Unix sockets.

### CODE REVIEW

File: lib/pure/net.nim [hunk 1]
```diff
@@ -582,6 +583,27 @@ proc connect*(socket: Socket, address: string,
       let ret = SSLConnect(socket.sslHandle)
       socketError(socket, ret)
 
+when defined(posix):
```
Comment (dom96) [inline lib/pure/net.nim]: ``or defined(nimdoc)`` please

File: lib/pure/net.nim [hunk 2]
```diff
@@ -582,6 +583,27 @@ proc connect*(socket: Socket, address: string,
...
+  proc makeUnixAddr(path: string): Sockaddr_un =
+    result.sun_family = AF_UNIX.toInt
+    if path.len >= Sockaddr_un_path_length:
+      raise newException(ValueError, "socket path too long")
+    copyMem(addr result.sun_path, path.cstring, path.len + 1)
+
+  proc connectUnix*(socket: Socket, path: string) =
+    ## Connects to Unix socket on `path`.
```
Comment (dom96) [inline lib/pure/net.nim]: Explain that this only works on Linux, Mac OS X and BSD operating systems in the doc comment.

File: lib/pure/net.nim [hunk 3]
```diff
@@ -582,6 +583,27 @@ proc connect*(socket: Socket, address: string,
...
+    var socketAddr = makeUnixAddr(path)
+    if socket.fd.connect(cast[ptr SockAddr](addr socketAddr),
+                      sizeof(socketAddr).Socklen) != 0'i32:
+      raiseOSError(osLastError())
+
+  proc bindUnix*(socket: Socket, path: string) =
+    ## Binds Unix socket to `path`.
```
Comment (dom96) [inline lib/pure/net.nim]: Same as above https://github.com/nim-lang/Nim/pull/3488/files#r56705284

**Comment by dom96** [comment]
Apologies about the extreme delay in reviewing this.

**Comment by girvo** [comment]
@dom96 Is the feedback you've given for this PR the only thing stopping it from being accepted? I'll gladly handle it: I've got a project that needs Unix socket support that I really want to do in Nim!

**Comment by dom96** [comment]
Yep. Go for it @girvo :)

**Pull request merged**